src/dbnode/storage/series/buffer.go

```diff
@@ -49,13 +50,21 @@ const (
 	// 3. Bucket for the future that can be taking writes that is head of
 	// the current block if write is for the future within bounds
 	bucketsLen = 3
+
+	// TODO: make sure this is a good pool size or make it customizable
+	defaultBufferBucketPoolSize = 16
```

Comment:
questioning whether pooling this at all is worth it, especially with a size of 16. I don't imagine we'll be creating these buckets so often that pooling will matter, and if we were, would probably want a global shared pool or something. Maybe Rob has some thoughts on this

Reply:
Yeah if it's only for buffers outside of the current realtime mutable buckets, we don't need a pool.

For the realtime ones that we rotate between constantly, if we keep that rotation buffer in here (i.e. bucketsRealtime) then we don't need to pool them since we will be reusing them anyway.